### DISCUSSION TITLE
How does webgpu planning to continue webgl shaders

### DISCUSSION THREAD

**Discussion by dellwatson** [gpuweb/gpuweb#4265] [Q&A] [answered]
since many vfx or particles are made from GLSL. 
for eg: [https://github.com/effekseer/Effekseer)](https://github.com/effekseer/Effekseer)

how do we port it into webgpu in the future? will that be automatically work on webgpu too? 
or people need re-create all of those in WGSL?

**Comment by Pauan** [comment, marked as answer]
You can use [naga](https://github.com/gfx-rs/naga) to convert from one shader format to another shader format (e.g. GLSL into WGSL).

**Reply by kainino0x** [reply]
Tint (part of [Dawn](https://dawn.googlesource.com/dawn/+/refs/heads/main/)) also has some shader conversion capabilities, in particular from SPIR-V to WGSL.
